pkgs/nix/default.nix

```diff
@@ -15,21 +15,32 @@
 # nixVersions.stable upstream break our build.
 #
 # ---------------------------------------------------------------------------- #
-{ nixVersions, stdenv }:
+{
+  lib,
+  nixVersions,
+  stdenv,
+}:
 let
-  # Temporarily pinning nix at 2.18 while we work to port pkgdb to latest
-  # stable (2.24), or replace it altogether.
-  nixVersion = "nix_2_18"; # "stable";
+  nixVersion = "stable";
 in
 nixVersions."${nixVersion}".overrideAttrs (prev: {
   # Necessary for compiling with debug symbols
   inherit stdenv;
 
   # Apply patch files.
-  patches = prev.patches ++ [
-    (builtins.path { path = ./patches/nix-9147.patch; })
-    (builtins.path { path = ./patches/multiple-github-tokens.2.13.2.patch; })
-  ];
+  patches =
+    prev.patches
+    ++ (
+      if lib.versionAtLeast prev.version "2.24" then
+        [
+          (builtins.path { path = ./patches/multiple-github-tokens.2.24.9.patch; })
+        ]
+      else
+        [
+          (builtins.path { path = ./patches/nix-9147.patch; })
```

Comment:
Is this omitted for newer versions of Nix because it was merged upstream?

Is it still needed against older versions of Nix?

Reply:
probably needed against older versions but i dont think we need to keep logic to build different versions of nix around when we only use one version anyway..
ill drop the <2.24 patches